### DISCUSSION TITLE
misorientation axes distribution by considering pairs of pixels located randomly

### DISCUSSION THREAD

**Discussion by Saoussen-ouhiba** [mtex-toolbox/mtex#637] [Ask Anything] [answered]
Hello,

I was able to draw the misorientation axis distribution for the grains list that I have by using the following script

```
gB=grains.boundary('Al','Al');
gBMis=gB.misorientation;
Misaxis=gBMis.axis;
Misangle=gBMis.angle./degree;
figure
plot(Misaxis,'fundamentalregion','contourf')
mtexColorbar
```
I would like now to draw the misorientation axis distribution by considering pairs of pixels selected randomly from the grain list. Could you help me to do it please?

Thank you a lot,

Saoussen

**Comment by kilir** [comment, marked as answer]
Hi,
while the grain list has no pixels, you can make a random selection of e.g. directions using `discreteSample`:
```
gmo = grains('Al').meanOrientation;
mo1 = discreteSample(gmo,100)
mo2 = discreteSample(gmo,100)
misAxisC = axis(mo1.\mo2);
```
You can also compute an mdf from an odf and either sample misorientations or use calcAxisDistribution from the mdf.

Cheers,
Rüdiger

**Reply by Saoussen-ouhiba** [reply]
Thank you so much Rüdiger. I am sorry for disturbing you again but I still have one more question about the function discreteSample(gmo,n). Does it mean that we are selecting randomly n mean grains orientations from gmo (not n pixels orientations from each grain)? And what function can I use if I do not want to get the same orientation twice (i.e. one time in mo1 and another time in mo2)?
Thank you a lot for your help,
Saoussen

**Reply by kilir** [reply]
```
mo1 = discreteSample(gmo,100)
```
does take 100 random samples with replacement from gmo. See also `help discreteSample`.

If you want all orientations, but in a random order you can either draw all without replacement or use `gmo(randperm(length(gmo)))` for example. 

Cheers,
Rüdiger

**Reply by Saoussen-ouhiba** [reply]
Thank you so much for your response.
